src/main/scala/beam/agentsim/agents/modalBehaviors/DrivesVehicle.scala

```diff
@@ -47,17 +47,12 @@ trait DrivesVehicle[T <: BeamAgentData] extends  TriggerShortcuts with HasServic
       passengerSchedule.schedule.get(completedLeg) match {
         case Some(manifest) =>
           _awaitingAlightConfirmation ++= manifest.alighters
-          manifest.riders.foreach(passenger => beamServices.vehicleRefs(passenger) ! NotifyLegEnd)
+          manifest.riders.foreach(pv => beamServices.personRefs.get(pv.personId).foreach( _  ! NotifyLegEnd))
           stay()
         case None =>
           log.error(s"Driver ${id} did not find a manifest for BeamLeg ${_currentLeg}")
           goto(BeamAgent.Error)
       }
-    case Event(ReservationRequestWithVehicle(req, vehicleIdToReserve), _) =>
-      require(passengerSchedule.schedule.nonEmpty, "Driver needs to init list of stops")
```

Comment:
This requirement is not general enough and highlights an issue with how ReservationRequest and handleVehicleReservation are currently working. When a Person makes a ReservationRequest, they should be passing some number of BeamLegs with the request, then we search the PassengerSchedule with the BeamLegs to determine whether there is sufficient capacity.

This process needs to work regardless of whether driving a Bus (which has a fixed schedule) or is a Taxi where the schedule is totally dynamic... so if the BeamLeg doesn't already exist in the PassengerSchedule, we should have an abstract method that validates whether it's ok to add that unrecognized BeamLeg.

Then we implement the logic of that method in the concrete class because the constraints around adding new BeamLegs will vary depending on the type of driver (e.g. a TransitDriver may never allow new BeamLegs but a PersonAgent may always allow since this is simply how we populate the PassengerSchedule with the data necessary to drive the vehicle).